src/main/java/com/github/tomakehurst/wiremock/http/HttpClientFactory.java

```diff
@@ -66,14 +70,27 @@ public static HttpClient createClient(
         return builder.build();
 	}
 
-    private static SSLContext buildSSLContextWithTrustStore(KeyStoreSettings trustStoreSettings) {
+    private static SSLContext buildSSLContextWithTrustStore(KeyStoreSettings trustStoreSettings, final String trustStoreKeyAlias) {
         try {
             KeyStore trustStore = trustStoreSettings.loadStore();
-            return SSLContexts.custom()
-                    .loadTrustMaterial(null, new TrustSelfSignedStrategy())
-                    .loadKeyMaterial(trustStore, trustStoreSettings.password().toCharArray())
-                    .useTLS()
-                    .build();
+            if (trustStoreKeyAlias != null) {
+              return SSLContexts.custom()
+                .loadTrustMaterial(null, new TrustSelfSignedStrategy())
+                .loadKeyMaterial(trustStore, trustStoreSettings.password().toCharArray(), new PrivateKeyStrategy() {
+                  @Override
+                  public String chooseAlias(Map<String, PrivateKeyDetails> aliases, Socket socket) {
+                    return trustStoreKeyAlias;
+                  }
+                })
+                .useTLS()
+                .build();
+            } else {
+              return SSLContexts.custom()
+                .loadTrustMaterial(null, new TrustSelfSignedStrategy())
+                .loadKeyMaterial(trustStore, trustStoreSettings.password().toCharArray())
+                .useTLS()
+                .build();
+            }
```

Comment:
@tomakehurst  please recommend an appropriate test case for this and I will add it in.